vimdoc/module.py

```diff
@@ -141,25 +142,35 @@ def Close(self):
     if ((self.GetCollection(vimdoc.FLAG) or
          self.GetCollection(vimdoc.SETTING)) and
         'config' not in self.sections):
-      config = Block()
-      config.SetType(vimdoc.SECTION)
+      config = Block(vimdoc.SECTION)
       config.Local(id='config', name='Configuration')
       self.Merge(config)
-    if not self.order:
-      self.order = []
-      for builtin in [
-          'intro',
-          'config',
-          'commands',
-          'autocmds',
-          'settings',
-          'dicts',
-          'functions',
-          'exceptions',
-          'mappings',
-          'about']:
-        if builtin in self.sections or builtin in self.backmatters:
-          self.order.append(builtin)
+
+    # Use explicit order as partial ordering and merge with default section
+    # ordering. All custom sections must be ordered explicitly.
+    # Custom sections will be ordered after all sections besides 'about'.
+    order = self.order or []
+    builtins_pre = []
+    for builtin in [
+        'intro',
+        'config',
+        'commands',
+        'autocmds',
+        'settings',
+        'dicts',
+        'functions',
+        'exceptions',
+        'mappings']:
+      if builtin in self.sections or builtin in self.backmatters:
+        if builtin not in order:
+          builtins_pre.append(builtin)
+    order = builtins_pre + order
```

Comment:
Don't think so. It only adds if it's not already there (`if builtin not in order:`).

I struggled with making the ordering smarter so it could stay closer to the default ordering. For instance, if you had:
`@order intro custom1 custom2 config about` and also defined sections for commands and functions, this implementation would give you
  `commands functions intro custom1 custom2 config about`
instead of
  `intro custom1 custom2 config commands functions about`
I had trouble even coming up with the words to describe the better behavior clearly.

If that order feels more intuitive, I can try to update it before submitting.